lib/exp.py

```diff
@@ -165,6 +165,9 @@ def create_job(self, experiment, parameters):
                 'experiment_uid': experiment.uid(),
                 'experiment_name': experiment.name
             },
+            'annotations': {
+                'experiment-job-parameters': json.dumps(parameters)
```

Comment:
Other metadata keys are using more pythonic `underscore_separated_strings`, but this one is using `hypen-separated-strings`.  For consistency we should pick one style and use it throughout